### DISCUSSION TITLE
Unable to open in Jupyterlab

### DISCUSSION THREAD

**Discussion by pamelafox** [community/community#58720] [Codespaces]
### Select Topic Area

Bug

### Body

I'm using the Python feature with installJupyterLab, but my codespaces can't be opened in Jupyterlab. 

To replicate, create a codespace from one of these links:
https://github.com/pamelafox/materials-fa22/blob/main/links.md

Then try to open in Jupyterlab.

Any idea what's the issue?

**Comment by pamelafox** [comment]
Update: I'm going back to a previous configuration that worked, so I'll leave here the details for what didn't work:

Repo hash:

https://github.com/pamelafox/materials-fa22/tree/f06dea6fb9155921dabbb80a6ed6dae19538d5a0

devcontainer.json:

```
{
    "name": "Data 8: lab07",
    "build": {
        "dockerfile": "../dockerfile",
        "context": "../.."
    },
    "features": {
        "ghcr.io/devcontainers/features/python:1": {
            "version": "3.11",
            "installJupyterlab": true
        }
    },
    "customizations": {
        "codespaces": {
            "openFiles": []
        },
        "vscode": {
            "settings": {
                "python.defaultInterpreterPath": "/usr/local/bin/python",
                "workbench.editorAssociations": {
                    "*.ipynb": "jupyter-notebook"
                },
                "notebook.compactView": false
            },
            "extensions": [
                "ms-python.python",
                "ms-toolsai.jupyter",
                "MS-vsliveshare.vsliveshare"
            ]
        }
    },
    "postCreateCommand": "python3 -m pip install -r .devcontainer/requirements.txt",
    "postAttachCommand": "code /workspaces/materials-fa22/materials/fa22/lab/lab07/lab07.ipynb"
}
```

dockerfile:

```
FROM mcr.microsoft.com/devcontainers/base:bullseye
```

**Reply by dmgardiner25** [reply]
@pamelafox ~~What was the diff between the new configuration that wasn't working and the old one that does?~~ Are you saying adding `"installJupyterlab": true` caused it to not work?

**Reply by pamelafox** [reply]
The two configs are fairly different:

https://github.com/pamelafox/materials-fa22/commit/5c23f815a7c43850941f70c8a62d01b6fa83c48d

The non-working one uses mcr.microsoft.com/devcontainers/base:bullseye as the base image, and then installs Python and Jupyterlab using the features. The working one uses mcr.microsoft.com/devcontainers/python:3.10 as the base image and installs Jupyterlab in the Dockerfile using requirements.txt. I've tried a few other configurations that don't work besides that, however. 

I'll see if I can figure out a more minimal difference, I was just hoping that you might have error logs on your side showing the error.

**Reply by pamelafox** [reply]
I've tried to make more minimal examples for you, so I started from the Codespaces Jupyter template, and then tried various base images plus the Jupyterlab feature. You can see my sequence of commits:
https://github.com/github/codespaces-jupyter/compare/main...pamelafox:codespaces-jupyter:main

The only thing that opened successfully in JupyterLab was a base image of universal, with or without the python/installJupyterlab feature. None of the other base images were openable.

My goal is to create a Codespace that has a *single* Python version but can also be opened in JupyterLab, hence all my experimentation with base images.
